src/components/Calendar/context/ContextWrapper.tsx

```diff
@@ -1,111 +1,123 @@
-    const[monthIndex,setMonthIndex] = useState(dayjs().month());
-    const[weekOffset,setWeekOffset] = useState(0);
-    const[dateOffset,setDateOffset]=useState(0);
-    const[Current_view,SetCurrent_view] = useState<string>("Month")
-    const[showEventModal,setShowEventModal] = useState(false);
-    const [daySelected, setDaySelected] = useState(dayjs());
-    const [selectedEvent, setSelectedEvent] = useState<selectedDayEvent | null>(null);
-    const[labels,setLabels] = useState<updateLabelAction[]>([])
-    const [savedEvents,dispatchCallEvents] = useReducer(savedEventsReducer,[],initEvents);
-    const [timeFrom, setTimeFrom] = useState<string|null>(selectedEvent ? selectedEvent.timeFrom : "from");
-    const [timeTo, setTimeTo] = useState<string|null>(selectedEvent ? selectedEvent.timeTo : "to");
+export default function ContextWrapper(props:any) {
+  const [monthIndex, setMonthIndex] = useState(dayjs().month());
+  const [weekOffset, setWeekOffset] = useState(0);
+  const [dateOffset, setDateOffset] = useState(0);
+  const [Current_view, SetCurrent_view] = useState('Month');
+  const [showEventModal, setShowEventModal] = useState(false);
+  const [daySelected, setDaySelected] = useState(dayjs());
+  const [selectedEvent, setSelectedEvent] = useState(null);
+  const [labels, setLabels] = useState(() => {
+    return Array.from(labelsClasses.keys()).map(label => ({
+        label,
+        checked: true
+    }));
+});
+  const [savedEvents, dispatchCallEvents] = useReducer(savedEventsReducer, []);
+  const [timeFrom, setTimeFrom] = useState('from');
+  const [timeTo, setTimeTo] = useState('to');
 
-    const filteredEvents = useMemo(() => {
-      return savedEvents.filter((evt:any) => 
-        labels.filter((lbl:updateLabelAction) => lbl.checked)
-        .map((lbl:updateLabelAction) => lbl.label)
-        .includes(evt.label)
-        );
-    },[savedEvents,labels]);
 
+const filteredEvents = useMemo(() => {
+  return savedEvents.filter((evt) =>
+      labels
+        .filter((lbl) => lbl.checked)
+        .map((lbl) => lbl.label)
+        .includes(evt.type)
+    );
+}, [savedEvents, labels]);
 
-    useEffect(() => {
-      setLabels((prevLabels) => {
-        return [...new Set(savedEvents.map((evt:updateLabelAction) => evt.label))].map((label:any) =>
-        {
-          const currentLabel = prevLabels.find(lbl => lbl.label === label)
-          return {
-            label,
-            checked : currentLabel ? currentLabel.checked :true,
-          }
-        })
-      })
-    },[savedEvents])
-    
-    useEffect(() => {
-      localStorage.setItem("savedEvents",JSON.stringify(savedEvents));
-    },[savedEvents])
+  const backendUrl = process.env.NEXT_PUBLIC_BACKEND_URL;
 
-    useEffect(() => {
-      if(!showEventModal){
-        setSelectedEvent(null);
+  useEffect(() => {
+    async function fetchEvents() {
+      try {
+        const response = await fetch(`${backendUrl}/api/v1/events`);
```

Comment:
don't fetch it here
write it in helpers/api and use the apiCall function

Reply:
done changes